### PR TITLE
Test that can reject federated invites

### PR DESCRIPTION
**Pull request by erikjohnston**
Tests matrix-org/synapse#12409

### CODE REVIEW

File: lib/SyTest/Homeserver.pm [hunk 1]
```diff
@@ -241,7 +241,7 @@ sub configure_logger
 
       loggers => {
          synapse => {
-            level => "INFO"
+            level => "DEBUG"
```
Comment (clokep) [inline lib/SyTest/Homeserver.pm:244]: I suspect you didn't want to include this?

File: tests/31sync/15lazy-members.pl [hunk 2]
```diff
@@ -807,3 +807,51 @@ sub check_filler_event {
          Future->done(1);
       });
    };
+
+
+test "Rejecing invite over federation doesn't break incremental /sync",
```
Comment (clokep) [inline tests/31sync/15lazy-members.pl:812]:
```suggestion
test "Rejecting invite over federation doesn't break incremental /sync",
```

File: tests/31sync/15lazy-members.pl [hunk 3]
```diff
@@ -807,3 +807,51 @@ sub check_filler_event {
...
+
+   check => sub {
+      my ( $invitee, $creator, $room_id ) = @_;
+
+      my ( $filter_id );
+
+      matrix_create_filter( $invitee, {
+         room => {
+            state => {
+               lazy_load_members => JSON::true,
```
Comment (clokep) [inline tests/31sync/15lazy-members.pl:828]: Does this really require lazy loading? I'm not really sure I'm following that bit.

File: lib/SyTest/Homeserver.pm (hunk 1 above)
Comment (erikjohnston) [reply lib/SyTest/Homeserver.pm:244]: /me grumbles at muscle memory and `-a`

File: tests/31sync/15lazy-members.pl (hunk 3 above)
Comment (erikjohnston) [reply tests/31sync/15lazy-members.pl:828]: During lazy loading we pull out state a second time to get the membership event of the sender of the outlier event, so we stick lazy loading on to hit more of the /sync code paths

Comment (clokep) [reply tests/31sync/15lazy-members.pl:828]: Thanks! 👍

**Review by clokep — APPROVED** [review]

**Comment by clokep** [comment]
(Well approved with the debug change...)

File: lib/SyTest/Homeserver.pm (hunk 1 above)
Comment (clokep) [reply lib/SyTest/Homeserver.pm:244]:
```suggestion
            level => "INFO"
```

**Pull request merged**